wait.go

```diff
@@ -352,20 +353,18 @@ func (c *Client) IsRedisFailoverReady(item *unstructured.Unstructured) (bool, st
 }
 
 func (c *Client) IsPostgresqlReady(item *unstructured.Unstructured) (bool, string) {
-	name := item.GetName()
-	namespace := item.GetNamespace()
+	status := item.Object["status"]
 
-	clientset, err := c.GetClientset()
-	if err != nil {
-		return false, fmt.Sprintf("failed to get clientset: %v", err)
+	if status == nil {
+		return false, "⏳ waiting to become ready"
 	}
 
-	sts, err := clientset.AppsV1().StatefulSets(namespace).Get(context.TODO(), name, metav1.GetOptions{})
-	if err != nil {
-		return false, fmt.Sprintf("failed to get sts %s: %v", name, err)
+	state := item.Object["status"].(map[string]interface{})["PostgresClusterStatus"]
+	if state != "Running" {
+		return false, "⏳ waiting to become ready"
 	}
 
-	return IsStatefulSetReady(sts)
+	return true, ""
```

Comment:
I think we wanted to explicitly check the health of underlying statefulset due lag / sync issues updating the CRD - we should probably add it as a commment and TODO to switch back to CRD readiness once it is less flakey

Reply:
In my testing, the postgresql won't be ready after the statefulset is ready (the postgres cluster hasn't initialized yet and thus not accepting any incoming requests) so I have to switch to checking CR's status which is much more accurate and reliable (because Zalando operator actually queries for the status of postgres and update the CR)

Not related to this but I found a flaw in the implementation of `IsDeploymentReady` and `IsStatefulSetReady` and I just pushed a fix.

Reply:
We have found in production the CR status to be unreliable - I think a combination of STS isready + checking that there is at least 1 pod with spilo-role=master, and all other pods are spilo-role=replica